flo-workflow/src/main/java/com/spotify/flo/Values.java

```diff
@@ -63,6 +64,9 @@ private Values() {
     BiConsumer<T, Throwable> firstComplete = (t, firstThrowable) -> {
       BiConsumer<U, Throwable> secondComplete = (v, secondThrowable) -> {
         if (firstThrowable != null) {
+          if (secondThrowable != null && secondThrowable != firstThrowable) {
+            firstThrowable.addSuppressed(secondThrowable);
```

Comment:
if `secondThrowable` is a cause or a suppressed exception of `firstThrowable` we end up with a loop here, but that's ok as `printStackTrace` and loggers etc should handle circular references in exceptions. I'd even remove the `secondThrowable != firstThrowable` check.

Reply:
The check is needed since addSuppressed throws if you add an axception to itself. This can happen if two tasks depend on the same task at two different levels.

There was actually a test failure that caught this.

Reply:
Since the two exceptions should come from different tasks, they should be independent. That’s why I ruled out any causal relationship between them.

The exception could however be the same exception if the same task is depended on in several tasks.